### PR TITLE
feat: services

### PR DESCRIPTION
**Pull request by ponderingdemocritus**
There was a bug within the services singelton pattern that cause the incorrect serivce to be called when multiple ones were used.

This changes the architecture so this doesn't happen.

### CODE REVIEW

File: packages/client-discord/src/messages.ts [hunk 1]
```diff
@@ -515,10 +515,17 @@ export class MessageManager {
...
                             // For voice channels, use text-to-speech
-                            const audioStream = await this.runtime
-                                .getService(ServiceType.SPEECH_GENERATION)
-                                .getInstance<ISpeechService>()
-                                .generate(this.runtime, content.text);
+
+                            const speechService =
+                                this.runtime.getService<ISpeechService>(
+                                    ServiceType.SPEECH_GENERATION
+                                );
```
Comment (monilpat) [inline packages/client-discord/src/messages.ts:522]: Can we add this:    if (!speechService) {
                    throw new Error("Speech generation service not found");
                }

File: packages/client-discord/src/voice.ts [hunk 2]
```diff
@@ -368,13 +369,12 @@ export class VoiceManager extends EventEmitter {
...
         let transcriptionStarted = false;
         let transcriptionText = "";
-        console.log("new audio monitor for: ", userId);
 
         const monitor = new AudioMonitor(
             audioStream,
             10000000,
             async (buffer) => {
-                console.log("buffer: ", buffer);
+                // console.log("buffer: ", buffer);
```
Comment (monilpat) [inline packages/client-discord/src/voice.ts:377]: Can we remove this thanks

File: packages/client-discord/src/voice.ts [hunk 3]
```diff
@@ -397,11 +397,14 @@ export class VoiceManager extends EventEmitter {
...
-                            .getService(ServiceType.TRANSCRIPTION)
-                            .getInstance<ITranscriptionService>()
-                            .transcribe(wavBuffer);
+                        const transcriptionService =
+                            this.runtime.getService<ITranscriptionService>(
+                                ServiceType.TRANSCRIPTION
+                            );
+
+                        const text =
+                            await transcriptionService.transcribe(wavBuffer);
```
Comment (monilpat) [inline packages/client-discord/src/voice.ts:406]: if (!transcriptionService) {
                                    throw new Error(
                                        "Transcription generation service not found"
                                    );
                                }

**Review by monilpat — CHANGES_REQUESTED** [review]
Great job with this feature! Thanks for getting this out! LGTM outside of addressing comments :)

**Comment by monilpat** [comment]
Great work here! Feel free to a quick TLDR using the PR template if you can on the issue and any testing plan thank you so much!

**Comment by ponderingdemocritus** [comment]
found some issues still - working on them

**Pull request merged**